### PR TITLE
Rename rustc guide

### PR DESCRIPTION
**Pull request by spastorino**
This is in preparation for https://github.com/rust-lang/rustc-guide/issues/470
Needs to be merged after we actually rename the guide.

Have used this to rename:

`git grep -l 'rustc_guide' | xargs sed -i 's/rustc_guide/rustc_dev_guide/g'`
`git grep -l 'rustc-guide' | xargs sed -i 's/rustc-guide/rustc-dev-guide/g'`
`git grep -l 'rustc guide' | xargs sed -i 's/rustc guide/rustc dev guide/g'`

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -31,9 +31,9 @@
 [submodule "src/stdarch"]
 	path = src/stdarch
 	url = https://github.com/rust-lang/stdarch.git
-[submodule "src/doc/rustc-guide"]
-	path = src/doc/rustc-guide
-	url = https://github.com/rust-lang/rustc-guide.git
+[submodule "src/doc/rustc-dev-guide"]
+	path = src/doc/rustc-dev-guide
+	url = https://github.com/rust-lang/rustc-dev-guide.git
```
Comment (spastorino) [inline .gitmodules]: Is this change handled gracefully by our build system or do I need to change something else? cc @Mark-Simulacrum

Comment (mark-i-m) [reply .gitmodules]: I think it's fine. But perhaps also git mv to rename the directory.

Comment (spastorino) [reply .gitmodules]: We should update the submodule after the rename. As soon as we do I can fix this PR.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -434,7 +434,7 @@ For people new to Rust, and just starting to contribute, or even for
 more seasoned developers, some useful places to look for information
 are:
 
-* The [rustc guide] contains information about how various parts of the compiler work and how to contribute to the compiler
+* The [rustc dev guide] contains information about how various parts of the compiler work and how to contribute to the compiler
```
Comment (LeSeulArtichaut) [inline CONTRIBUTING.md]: Makes me wonder if we should decide whether to use `rustc-dev-guide` or `rustc dev guide` and stick to one of them, or if it is completely insignificant

File: src/librustc_ast/README.md [hunk 3]
```diff
@@ -3,7 +3,7 @@ The `rustc_ast` crate contains those things concerned purely with syntax
 lexer, macro expander, and utilities for traversing ASTs.
 
 For more information about how these things work in rustc, see the
-rustc guide:
+rustc dev guide:
```
Comment (LeSeulArtichaut) [inline src/librustc_ast/README.md]: Maybe add an active link here?

**Review by LeSeulArtichaut — APPROVED** [review]
Looks good to me, haven't spotted any mistake.

File: src/librustc_ast/README.md (hunk 3 above)
Comment (spastorino) [reply src/librustc_ast/README.md]: I think we should, but maybe let's handle that on a separate PR?

File: CONTRIBUTING.md (hunk 2 above)
Comment (spastorino) [reply CONTRIBUTING.md]: well I was just naively translating things, unsure if it's significant but I guess we can discuss on a separate PR.

**Comment by JohnTitor** [comment]
36003a2 includes clippy sudmodule, we should drop it.

**Review by mark-i-m — CHANGES_REQUESTED** [review]
Mostly looks good to me, but it looks like you accidentally updated the clippy submodule.

**Comment by spastorino** [comment]
Seems like the repo was dirty because the command I was running doesn't include submodules. Anyway, fixed thanks for pointing that out.

**Review by mark-i-m — APPROVED** [review]
Looks good now!

**Comment by JohnTitor** [comment]
Actually, we have to rename the repo actually to r+, let's hold off on at this point.
r? @JohnTitor

**Comment by spastorino** [comment]
Yeah, this one is not ready. I thought I was changing rustc-guide :P. We need to first rename the repo.

File: .gitmodules (hunk 1 above)
Comment (Mark-Simulacrum) [reply .gitmodules]: I believe this should be fine, though I'm personally uncertain whether something else needs to change (i.e., our tooling should be fine but I don't know what git'll want).

**Comment by spastorino** [comment]
This should be now, ready to merge

**Comment by JohnTitor** [comment]
r=me on behalf of the WG once CI is happy.

**Comment by JohnTitor** [comment]
Ah, it seems the previous submodule remains: https://github.com/rust-lang/rust/tree/f4445da203ca08659f513a841f13cbfbb22b460b/src/doc

**Comment by spastorino** [comment]
fixed!

**Review by mark-i-m — APPROVED** [review]
LGTM

**Comment by pietroalbini** [comment]
r=me once CI passes.

**Pull request merged**